test/lib/rubycritic/report_generators/global_rating_calculator_test.rb

```diff
@@ -0,0 +1,15 @@
+require "test_helper"
+require "rubycritic/generators/global_rating_calculator"
+require "rubycritic/core/analysed_module"
+
+describe Rubycritic::Generator::GlobalRatingCalculator do
+  it "calculates the gpa" do
+    smells1 = [SmellDouble.new(:cost => 1), SmellDouble.new(:cost => 2)]
+    smells2 = [SmellDouble.new(:cost => 10), SmellDouble.new(:cost => 12)]
+    am1 = Rubycritic::AnalysedModule.new(:smells => smells1, :complexity => 0, :lines => 25)
+    am2 = Rubycritic::AnalysedModule.new(:smells => smells2, :complexity => 0, :lines => 10)
+    analysed_modules = [am1, am2]
+    gpa = Rubycritic::Generator::GlobalRatingCalculator.calculate_gpa(analysed_modules)
```

Comment:
The test set up here is huge. Normally I'd suggest to introduce factories here but I guess this would be a bigger topic that should be adressed in an separate pull request.